File: web/src/types/notification.ts
import type { Pagination } from "@/types/api";

export interface UserNotification {
  id: number;
  title: string;
  body: string;
  severity: string;
  data?: Record<string, string> | null;
  created_at: number;
  read_at?: number | null;
  is_read: boolean;
}

export interface NotificationsResponse {
  data: UserNotification[];
  pagination: Pagination;
}

export interface UnreadNotificationsResponse {
  unread_count: number;
}

File: mobile/app/src/main/java/org/yooud/airsense/models/Notification.kt
package org.yooud.airsense.models

import com.google.gson.annotations.SerializedName

data class UserNotification(
    val id: Long,
    val title: String,
    val body: String,
    val severity: String,
    val data: Map<String, String>?,
    @SerializedName("created_at")
    val createdAt: Long,
    @SerializedName("read_at")
    val readAt: Long?,
    @SerializedName("is_read")
    val isRead: Boolean
)

data class UnreadNotifications(
    @SerializedName("unread_count")
    val unreadCount: Int
)

